tests/Unit/ParallelAlgorithms/Actions/Test_SetData.cpp

```diff
@@ -0,0 +1,50 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <string>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Parallel/PhaseDependentActionList.hpp"
+#include "ParallelAlgorithms/Actions/SetData.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/ActionTesting.hpp"
+
+namespace {
+
+struct SomeNumber : db::SimpleTag {
+  using type = int;
+};
+
+template <typename Metavariables>
+struct Component {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = int;
+  using phase_dependent_action_list = tmpl::list<Parallel::PhaseActions<
+      typename Metavariables::Phase, Metavariables::Phase::Initialization,
+      tmpl::list<ActionTesting::InitializeDataBox<tmpl::list<SomeNumber>>>>>;
+};
+
+struct Metavariables {
+  using component_list = tmpl::list<Component<Metavariables>>;
```

Comment:
Another newbie question. How is this not infinite recursion, putting the name of the struct containing component_list into component_list? I'd love to better understand how this line works

Reply:
If I understand correctly, type aliases are lazily instantiated, so only when `::component_list` is retrieved will the `Component<Metavariables>` be instantiated. I think this means we need to be careful to avoid recursion when using the alias within the class (only if it’s a class template?), but using it from outside the class is fine.. again, e.g. @nilsdeppe or @wthrowe can confirm or correct me here.